background-charm-service/src/worker.ts

```diff
@@ -5,14 +5,14 @@ import {
   isStream,
   onError,
   setBobbyServerUrl,
+  setRecipeEnvironment,
```

Comment:
could this be `setCharmEnvironment` - and we expose the charmId inside the context as well?

Reply:
We could, though a recipe environment is static for a runtime, and I imagine a charm environment is unique to each charm, which would be tedious to manage with globals. There's probably a better way to inject run-specific metadata